library/ssl_tls13_client.c

```diff
@@ -1011,7 +1011,7 @@ int mbedtls_ssl_write_pre_shared_key_ext( mbedtls_ssl_context *ssl,
          */
         if( ssl->handshake->early_data == MBEDTLS_SSL_EARLY_DATA_ON )
         {
-            ssl->session_negotiate->key_exchange = MBEDTLS_SSL_TLS13_KEY_EXCHANGE_MODE_PSK_KE;
+            ssl->session_negotiate->key_exchange = MBEDTLS_KEY_EXCHANGE_NONE;
```

Comment:
I agree that we have to distinguish key exchange and PSK mode, but none doesn't seem correct here. It could be either `MBEDTLS_KEY_EXCHANGE_PSK` or `MBEDTLS_KEY_EXCHANGE_ECDHE_PSK`. 

I think we should take a look at where `session_negotiate->key_exchange` actually matters.

If this is not a blocker, feel free to create an issue for this - I recently created another one related to PSK modes, so one could aim to address both in one go.

Reply:
open #120 for it.